xformers/components/attention/nystrom.py

```diff
@@ -194,8 +194,9 @@ def forward(
             mask: Optional[torch.Tensor] = None
 
             if self.causal:
-                mask = self._tril_mask(batched_dim, seq_len, seq_len, **tt)
+                mask = self._triu_mask(batched_dim, seq_len, seq_len, **tt)
 
+            # FIXME: @lefaudeux This is probably broken, we don´t test this I suppose
```

Comment:
This should be tested in `test_masking` in `test_nystrom_attention.py`

Reply:
it's this line which I thought could not possibly work, given the difference in shapes ?

Reply:
I think it should work since we assert on `key_padding_mask` dimension being (batch, 1, seq)? And we also convert it to an additive mask. But perhaps using `masked_fill` with a boolean key padding mask makes more sense?

Reply:
yes you're right, sorry I forgot again, the 1 will be enough to broadcast ! all good, 'll remove the FIXME, I've changed the unit test to make sure that we do go there and it works indeed